OpenSim/Actuators/VandenBogert2011Muscle.h

```diff
@@ -0,0 +1,568 @@
+        void setDefaultFiberLength(double fiberLength);
+        /** @returns the default muscle fiber length,, ()*/
+        double getDefaultFiberLength() const;
+
+
+        /**@param s The state of the system.
+        @param the muscle activation, (dimensionless)*/
+        void setActivation(SimTK::State& s,double activation) const override;
+        /**@param s The state of the system.
+        @param fiberLength the muscle fiber length, ()*/
+        void setFiberLength(SimTK::State& s,double fiberLength) const;
+
+        /* Calculate the muscle implicit residual.  Returns a state Vec2
+        containing the force residual and activation residual.  When these
+        value are equal to zero, the muscle equations are "balanced" and the
+         muscle  is muscle is in a valid state.
+                @param s The state of the system.
+                @param projFibVelNorm_guess The muscle fiber velocity, projected
+                in line with tendon and normalized by the optimal fiber length,
+                (dimensionless)
+                @param activdot_guess The muscle activation time derivative,
+                (1/s)
+                @returns a vector of the muscle residual:
+                        \li 0: force (N)
+                        \li 1: activation (unitless)*/
+        SimTK::Vec2 getResidual(const SimTK::State& s,
+                                double projFibVelNorm_guess,
+                                double activdot_guess, double u) const;
+
+
+        //Hacks because cache variables not implemented yet
+
+        /**@param The state of the system.
+           @returns The Fiber Length of the muscle.*/
+        double  getFiberLength(SimTK::State& s) const;
+        /**@param The state of the system.
+           @returns The activation of the muscle.*/
+        double  getActivation(SimTK::State& s) const;
+
+
+        //Temporary struct for troubleshooting
+        struct ImplicitResidual {
+            double forceResidual = SimTK::NaN;
+            double activResidual = SimTK::NaN;
+            double forceTendon = SimTK::NaN;
+            SimTK::Mat22 df_dy = {SimTK::NaN,SimTK::NaN,SimTK::NaN,
+                                  SimTK::NaN,SimTK::NaN,SimTK::NaN};
+            SimTK::Mat22 df_dydot = {SimTK::NaN,SimTK::NaN,SimTK::NaN,
+                                     SimTK::NaN,SimTK::NaN,SimTK::NaN};
+            SimTK::Vec2 df_du = {SimTK::NaN,SimTK::NaN};
+            double df_dmuscleLength = SimTK::NaN;
+            double F1 = SimTK::NaN;
+            double F2 = SimTK::NaN;
+            double F3 = SimTK::NaN;
+            double F4 = SimTK::NaN;
+            double F5 = SimTK::NaN; };
+
+//==============================================================================
+// MUSCLE.H INTERFACE
+//==============================================================================
+        /** @param[in] s The state of the system.
+            @returns The tensile force the muscle is generating (N). */
```

Comment:
I think you should leave this doxygen comment in, because it is good. But in general, for methods you are overriding, you don't need a doxygen comment unless you want to say something specific about this class's implementation of the virtual method.